app/uk/gov/hmrc/selfservicetimetopay/controllers/ArrangementController.scala

```diff
@@ -168,25 +161,21 @@ class ArrangementController(ddConnector: DirectDebitConnector,
     Some(ttpSubmission.calculatorData.copy(firstPaymentDate = Some(firstPaymentDate), endDate = lastPaymentDate))
   }
 
-  def submit(): Action[AnyContent] = AuthorisedSaUser {
-    implicit authContext => implicit request =>
+  def submit(): Action[AnyContent] = AuthorisedSaUser { implicit authContext =>
+    implicit request =>
       authorizedForSsttp {
-        sessionCache.get.flatMap {
-          _.fold(redirectToStart)(arrangementSetUp)
-        }
+        case None => redirectToStart
+        case Some(ttp) => arrangementSetUp(ttp)
       }
   }
 
-  def applicationComplete(): Action[AnyContent] = AuthorisedSaUser {
-    implicit authContext => implicit request =>
+  def applicationComplete(): Action[AnyContent] = AuthorisedSaUser { implicit authContext =>
+    implicit request =>
       authorizedForSsttp {
-        sessionCache.get.flatMap {
-          _.fold(redirectToStart)(submission => {
-            sessionCache.remove()
-            successful(Ok(application_complete(submission.taxpayer.get.selfAssessment.get.debits.sortBy(_.dueDate.toEpochDay()),
-              submission.arrangementDirectDebit.get, submission.schedule.get, loggedIn = true)))
-          })
-        }
+        case None => redirectToStart
+        case Some(submission) =>
+          sessionCache.remove().map(_ => Ok(application_complete(submission.taxpayer.get.selfAssessment.get.debits.sortBy(_.dueDate.toEpochDay()),
```

Comment:
We weren't mapping on the result of the session cache removal - there was previously no guarantee the cached data was cleared before this returned